azure-iot-device/tests/iothub/pipeline/test_pipeline_stages_iothub.py

```diff
@@ -668,6 +683,86 @@ def test_does_not_change_last_version_seen_if_same_version(
         assert stage.last_version_seen == new_version
 
 
+@pytest.mark.describe(
+    "EnsureDesiredPropertiesStage - OCCURRENCE: GetTwinOperation that was sent down by this stage completes and pipeline is NOT connected"
+)
+class TestEnsureDesiredPropertiesStageWhenGetTwinOperationCompletesNotConnected(
+    EnsureDesiredPropertiesStageTestConfig
+):
+    @pytest.fixture
+    def stage(self, mocker, cls_type, init_kwargs, nucleus):
+        stage = cls_type(**init_kwargs)
+        stage.nucleus = nucleus
+        stage.nucleus.pipeline_configuration.ensure_desired_properties = True
+        stage.send_op_down = mocker.MagicMock()
+        stage.send_event_up = mocker.MagicMock()
+        mocker.spy(stage, "report_background_exception")
+        return stage
+
+    @pytest.fixture
+    def get_twin_op(self, stage):
+        stage.last_version_seen = -1
+        stage.handle_pipeline_event(pipeline_events_base.ConnectedEvent())
+
+        get_twin_op = stage.send_op_down.call_args[0][0]
+        assert isinstance(get_twin_op, pipeline_ops_iothub.GetTwinOperation)
```

Comment:
Yes, perfect. For extra paranoia, I would assert the value of  `send_op_down.call_count` before and after the `ConnectedEvent` to make sure we're working with the `GetTwinOperation` that came from the `ConnectedEvent`

Reply:
Good suggestion, thanks!